stripe/api_resources/list_object.py

```diff
@@ -138,23 +142,25 @@ def next_page(
 
     def previous_page(
         self, api_key=None, stripe_version=None, stripe_account=None, **params
-    ):
-        if not self.has_more:  # type: ignore
+    ) -> Self:
+        if not self.has_more:
             return self.empty_list(
                 api_key=api_key,
                 stripe_version=stripe_version,
                 stripe_account=stripe_account,
             )
 
-        first_id = self.data[0].id  # type: ignore
+        first_id = self.data[0].id
 
         params_with_filters = self._retrieve_params.copy()
         params_with_filters.update({"ending_before": first_id})
         params_with_filters.update(params)
 
-        return self.list(
+        result = self.list(
             api_key=api_key,
             stripe_version=stripe_version,
             stripe_account=stripe_account,
             **params_with_filters
         )
+        assert isinstance(result, ListObject)
```

Comment:
We already did this for next_page but the previous_page analog got lost :(. Would have been nice to go in the major since it can technically break somebody testing against unrealistic mocks, but can't break actual requests against the API, so I think it's peaceful to ship now.